modyn/supervisor/internal/triggers/utils.py

```diff
@@ -0,0 +1,140 @@
+
+    train_set = OnlineTriggerDataset(
+        dataloader_info.pipeline_id,
+        trigger_id,
+        dataloader_info.dataset_id,
+        dataloader_info.bytes_parser,
+        dataloader_info.transform_list,
+        dataloader_info.storage_address,
+        dataloader_info.selector_address,
+        dataloader_info.training_id,
+        dataloader_info.num_prefetched_partitions,
+        dataloader_info.parallel_prefetch_requests,
+        dataloader_info.tokenizer,
+        sample_prob,
+    )
+
+    logger.debug("Creating online trigger DataLoader.")
+    return DataLoader(train_set, batch_size=dataloader_info.batch_size, num_workers=dataloader_info.num_dataloaders)
+
+
+def prepare_trigger_dataloader_fixed_keys(
+    trigger_id: int,
+    dataloader_info: DataLoaderInfo,
+    keys: list[int],
+    sample_size: Optional[int] = None,
+) -> DataLoader:
+    if sample_size is not None:
+        keys = random.sample(keys, min(len(keys), sample_size))
+
+    train_set = FixedKeysDataset(
+        dataloader_info.dataset_id,
+        dataloader_info.bytes_parser,
+        dataloader_info.transform_list,
+        dataloader_info.storage_address,
+        trigger_id,
+        keys,
+        dataloader_info.tokenizer,
+    )
+
+    logger.debug("Creating fixed keys DataLoader.")
+    return DataLoader(train_set, batch_size=dataloader_info.batch_size, num_workers=dataloader_info.num_dataloaders)
+
+
+def get_embeddings(model_downloader: ModelDownloader, dataloader: DataLoader) -> torch.Tensor:
+    """
+    input: model_downloader with downloaded model
+    output: embeddings Tensor
+    """
+    assert model_downloader._model is not None
+    all_embeddings: Optional[torch.Tensor] = None
+
+    model_downloader._model.model.eval()
+    model_downloader._model.model.embedding_recorder.start_recording()
+
+    with torch.no_grad():
+        for batch in dataloader:
+            data: Union[torch.Tensor, dict]
+            if isinstance(batch[1], torch.Tensor):
+                data = batch[1].to(model_downloader._device)
+            elif isinstance(batch[1], dict):
+                data: dict[str, torch.Tensor] = {}  # type: ignore[no-redef]
+                for name, tensor in batch[1].items():
+                    data[name] = tensor.to(model_downloader._device)
+            else:
+                raise ValueError(f"data type {type(batch[1])} not supported")
+
+            with torch.autocast(model_downloader._device_type, enabled=model_downloader._amp):
+                model_downloader._model.model(data)
+                embeddings = model_downloader._model.model.embedding_recorder.embedding
+                if all_embeddings is None:
+                    all_embeddings = embeddings
+                else:
+                    all_embeddings = torch.cat((all_embeddings, embeddings), 0)
+
+    model_downloader._model.model.embedding_recorder.end_recording()
+
+    return all_embeddings
+
+
+def get_embeddings_evidently_format(
```

Comment:
I think it would be cleaner to pass the tensor here and have a convert tensor to evidently embedding function instead

Reply:
Removed get_embeddings_evidently_format(). Added convert_tensor_to_df(tensor, column_name_prefix).